src/main/java/de/filefighter/rest/domain/user/business/UserBusinessService.java

```diff
@@ -1,17 +1,108 @@
 package de.filefighter.rest.domain.user.business;
 
+import de.filefighter.rest.domain.token.data.dto.AccessToken;
+import de.filefighter.rest.domain.token.data.dto.RefreshToken;
+import de.filefighter.rest.domain.user.data.dto.User;
+import de.filefighter.rest.domain.user.data.persistance.UserEntity;
 import de.filefighter.rest.domain.user.data.persistance.UserRepository;
+import de.filefighter.rest.domain.user.exceptions.UserNotAuthenticatedException;
+import de.filefighter.rest.domain.user.exceptions.UserNotFoundException;
 import org.springframework.stereotype.Service;
 
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.sql.Ref;
+import java.util.Base64;
+import java.util.UUID;
+
+import static de.filefighter.rest.configuration.RestConfiguration.*;
+import static de.filefighter.rest.domain.common.Utils.stringIsValid;
+
 @Service
 public class UserBusinessService {
+
     private final UserRepository userRepository;
+    private final UserDtoService userDtoService;
 
-    public UserBusinessService(UserRepository userRepository) {
+    public UserBusinessService(UserRepository userRepository, UserDtoService userDtoService) {
         this.userRepository = userRepository;
+        this.userDtoService = userDtoService;
     }
 
-    public long getUserCount(){
+    public long getUserCount() {
         return userRepository.count();
     }
+
+    public User getUserByUsernameAndPassword(String base64encodedUserAndPassword) {
+        if (!stringIsValid(base64encodedUserAndPassword))
+            throw new UserNotAuthenticatedException("Header was empty.");
+
+        //TODO: maybe filter unsupported characters?
+        if (!base64encodedUserAndPassword.matches("^"+AUTHORIZATION_BASIC_PREFIX+"(.*)$"))
```

Comment:
braun would be proud

Reply:
actually i changed them to be even better. just didnt push